Automattic/happychat-service
# Adds handling for a 'reserve' operator status

This adds support for a new 'reserve' status for operators, and ensures operators with this status are only assigned after all 'available' operators are at max capacity.

For operators in reserve, priority of chat assignment is given to operators who already have a chat going, to avoid disturbing other operators until necessary.

### To test

 - Run `npm run test` and confirm that the following new tests and all existing tests pass:

```
  Chatlist Assignment
    ...
    ✓ should assign active operator before reserve operator
    ✓ should assign reserve operator when active operators are busy
    ✓ should assign to a reserve operator that is already chatting
    ✓ should assign to a different reserve operator when the first has reached two chats
    ✓ should assign evenly when all reserve operators have reached two chats
```

#### Todo
 - [x] Assign chats to reserve operators who already have chats, to avoid disturbing others
